### ISSUE TITLE
interpolation in shorthand properties

### ISSUE THREAD

**Issue by yairEO** [smooth-code/xstyled#64]
## theme

```
export const space = [0.8, 1, 2, 3, 4]
space.xs = space[0] / 2 + 'em'
space.small = space[0] + 'em'
space.medium = space[2] + 'em'
space.large = space[4] + 'em'
```

## Some styled component:

```
const StyleComp = styled.div`
	margin: 0 0 0 large;
`
```

**Undesired** output:

    margin: 0.8px 0.8px 0.8px 4em;

**Expected** output:

    margin: 0 0 0 4em;

----
 
I do understand the difficulty in this mixed input, where one would want `0` a real value and not place [0] in the theme "space" array. Are you planning to address this?

It's very annoying to specifically write it in 2 lines:

```
const StyleComp = styled.div`
	margin: 0;
    margin-left: large;
`
```

**Comment by neoziro** [comment]
The solution is simple, `0` has to be `0` if you want it to be `0`:

```js
export const space = [0, 0.8, 1, 2, 3, 4]
space.xs = space[1] / 2 + 'em'
space.small = space[1] + 'em'
space.medium = space[3] + 'em'
space.large = space[5] + 'em'
```

**Comment by yairEO** [comment]
I found out about this only after months of working with *xstyled*, and now i have tons of files with zeros, and I cannot simple modify the theme, adding `0` to the beginning of the `space` array, since tons of files are depended on the current indexes!

This is a huge problem for me now, because this simple yet **important** thing was not covered in the docs at the time when I first started using xtyled, months ago and now it's all over my workplace's system.

I need to think what to do about it. I have to find every place in the code that uses the words `space`/`small`/`medium`/`large` and manually see if there are any zeros there as well. 

So far I've manually changed every zeros I've found to `0px` (though it's not that pretty to look at IMO)
